src/docs/product/performance/queries.mdx

```diff
@@ -84,17 +102,34 @@ You can see the full query by hovering on a truncated description, or clicking i
 
 To open a query's **Query Summary** page, click on the query from either the table in the "Most Time-Consuming Queries" widget on the **Performance** page or the table in the **Queries** page.
 
-At the top of the page, queries per minute, average duration, and time spent are shown for the selected time range. The full query is shown below these metrics, followed by graphs for queries per minute and average duration.
+At the top of the page, queries per minute, average duration, and time spent are shown for the selected time range. The full query is shown below these metrics, followed by graphs for queries per minute and average duration. The query's source in the code is also shown if it is available.
 
 At the bottom, you can find a list of endpoints the query is found in, sorted by the most time your application spent in that span.
 
 If you want to investigate a specific endpoint, click on it to open a sidebar showing some sample events.
 
 The gif below provides a detailed walkthrough of the **Query Summary** page.
 
-<div style="position: relative; padding-bottom: calc(48.68055555555556% + 41px); height: 0; width: 100%"><iframe src="https://demo.arcade.software/MTzviZtIiN7ZCmNFVJg4?embed" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;color-scheme: light;" title="Performance - Queries - Query Summary Page Walkthrough"></iframe></div>
+<div style="position: relative; padding-bottom: calc(48.68055555555556% + 41px); height: 0; width: 100%">
+  <iframe
+    src="https://demo.arcade.software/MTzviZtIiN7ZCmNFVJg4?embed"
+    frameborder="0"
+    loading="lazy"
+    webkitallowfullscreen
+    mozallowfullscreen
+    allowfullscreen
+    style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;color-scheme: light;"
+    title="Performance - Queries - Query Summary Page Walkthrough"
+  ></iframe>
+</div>
+
+## Query Sources
 
-</br>
+A query's source is the location in your application's code that triggered the query. Sentry automatically determines the query source by examining the stack traceh and attaches the information to the query data. In order to reduce the SDK's performance impact on your application, Sentry only attaches query sources to queries that are slower than a <PlatformLink to="/configuration/options">configurable threshold</PlatformLink>. If query sources are often missing, you can reduce this threshold.
+
+You can see the query's source on the [event details page](/product/sentry-basics/concepts/tracing/event-detail/) next to other span properties in the span waterfall as well underneath the query description on the Query Summary page. If you have a [source code management integration](/product/integrations/source-code-mgmt/) enabled, Sentry will also show a link to open the query's source in the relevant repository.
```

Comment:
```suggestion
You can see the query's source underneath the query description on the **Query Summary** page, as well as on the [**Event Details** page](/product/sentry-basics/concepts/tracing/event-detail/) next to other span properties in the span waterfall. If you have a [source code management integration](/product/integrations/source-code-mgmt/) enabled, Sentry will also show a link to open the query's source in the relevant repository.
```
I would move the query summary page up first since it comes up first in the typical user flow.

If the source link is only shown on a specific page, I would include that info as well. But if shown in both event details & query summary pages, then ok as is.